### DISCUSSION TITLE
A clear example of additionalVariables object usage

### DISCUSSION THREAD

**Discussion by mikaelsand** [qetza/replacetokens-task#55] [Q&A] [answered]
In the documentation there are examples of how to supply additionalVariables using variables and multiple glob patterns but there is no example of the first "an object: properties will be parsed as key/value pairs". Can this be provided?

I am currently looking into having a powershell-script call into Azure to get metadata. This metadata will then be used to update a settings-file using the replacetokens-task. This is to avoid using variable groups for a lot of settings. Can this be done?

**Comment by qetza** [comment, marked as answer]
Hi @mikaelsand,
Thanks for the feedback, I'll add a example of passing additional variables using a object (map).

For your need, the task is able to automatically access environment variables (that's how the agent is passing inline variables and variable groups to tasks) and variables specified in the additional variables input. The additional variables are set in your definition so you cannot update them dynamically during runtime.

What you could do in your step where you get the metadata:
- dump the data in a json file and configure this file in the additional variables of the replace tokens task (and maybe add a step after to delete the file if it contains some sensitive data)
- or you can dynamically create variables so that they are available to the next steps using the command
  `Write-Host "##vso[task.setvariable variable=myVar;]$var"`
    (https://learn.microsoft.com/en-us/azure/devops/pipelines/process/set-variables-scripts?view=azure-devops&tabs=powershell#about-tasksetvariable)

Hope this helps.

**Reply by mikaelsand** [reply]
It helps a lot! Thank you.
